app/Exceptions/Handler.php

```diff
@@ -39,14 +40,31 @@ public function report(Exception $exception)
     /**
      * Render an exception into an HTTP response.
      *
-     * @param \Illuminate\Http\Request $request   the app request
-     * @param \Exception               $exception the exception that needed to be redered
+     * @param \Illuminate\Http\Request $request the app request
+     * @param \Exception               $e       the exception that needed to be redered
      *
      * @return \Illuminate\Http\Response
      */
-    public function render($request, Exception $exception)
+    public function render($request, Exception $e)
     {
-        return parent::render($request, $exception);
+        $viewName = $this->getViewName();
+        if ($e instanceof ModelNotFoundException) {
+            return response()->view($viewName);
+        }
+        return parent::render($request, $e);
+    }
+
+    /**
+     * Determine what view to show based on route
+     *
+     * @return string
+     */
+    protected function getViewName()
```

Comment:
this function name is not good. The purpose of this function is to get the view for 404 error page, but the function name doesn't have that meaning